dnf/rpm/__init__.py

```diff
@@ -45,6 +45,17 @@ def detect_releasever(installroot):
             msg = 'Error: rpmdb failed to list provides. Try: rpm --rebuilddb'
             raise dnf.exceptions.Error(msg)
         releasever = hdr['version']
+        try:
+            off = hdr[getattr(rpm, 'RPMTAG_PROVIDENAME')].index(distroverpkg)
+            flag = hdr[getattr(rpm, 'RPMTAG_PROVIDEFLAGS')][off]
+            ver  = hdr[getattr(rpm, 'RPMTAG_PROVIDEVERSION')][off]
+            if flag == rpm.RPMSENSE_EQUAL and ver:
+                if hdr['name'] != distroverpkg:
+                    # override the package version
+                    releasever = ver
+        except (ValueError, KeyError, IndexError):
```

Comment:
IMO only filtering by PROVIDENAME should be here (and with reduced number of exceptions.